### ISSUE TITLE
Dynamic properties in autocomplete JSON response

### ISSUE THREAD

**Issue by VMBindraban** [ampproject/amphtml#23814]
## Describe the new feature or change to an existing feature you'd like to see

Currently `autocomplete` requires the `items` property on the datasource as hardcoded [here](https://github.com/ampproject/amphtml/blob/54a576f763f0816d9e24185fbe314780dfec771d/extensions/amp-autocomplete/0.1/amp-autocomplete.js#L269).

My suggestion is to add an option to select the name of the property.

An other suggestion is support for `object` instead of an array like `amp-list` with `items="."` and `single-item`.

Reason for this change is that not everyone can change the remote datasource.

**Comment by VMBindraban** [comment]
Figured out that `amp-list` is the correct component for this.

**Comment by nainar** [comment]
In your case the link rel="canonical" is also pointing to the AMP page. It should be pointing to the non AMP page. 

I took a look at this page: https://informburo.kz/novosti/v-kyzylordinskom-sadike-detyam-davali-prosrochennye-i-protivopokazannye-lekarstva-.html
